lib/client/rpc/openbookMarkets.ts

```diff
@@ -0,0 +1,208 @@
+import { Program, Provider } from "@coral-xyz/anchor";
+import { OpenbookOrder, OpenbookProposalMarket } from "@/types/markets";
+import {
+  OpenbookV2,
+  baseLotsToUi,
+  priceLotsToUi,
+} from "@openbook-dex/openbook-v2";
+import { MarketAccount, OpenBookV2Client } from "@openbook-dex/openbook-v2";
+import { PublicKey } from "@solana/web3.js";
+import { getLeafNodes, getUsersOpenOrderPks } from "@/openbook";
+import { FutarchyMarketsClient } from "@/client";
+import { Proposal, TokenProps } from "@/types";
+
+export class FutarchyOpenbookMarketsRPCClient implements FutarchyMarketsClient {
+  private openbook: Program<OpenbookV2>;
+  private openbookClient: OpenBookV2Client;
+  private rpcProvider: Provider;
+  constructor(
+    rpcProvider: Provider,
+    openbook: Program<OpenbookV2>,
+    openbookClient: OpenBookV2Client
+  ) {
+    this.openbook = openbook;
+    this.openbookClient = openbookClient;
+    this.rpcProvider = rpcProvider;
+  }
+  async fetchConditionalMarketsFromProposal(
+    proposal: Proposal,
+    baseToken: TokenProps,
+    quoteToken: TokenProps
+  ): Promise<[OpenbookProposalMarket, OpenbookProposalMarket] | undefined> {
+    const accountInfos =
+      await this.rpcProvider.connection.getMultipleAccountsInfo([
+        proposal.account.openbookPassMarket,
+        proposal.account.openbookFailMarket,
+        proposal.account.openbookTwapPassMarket,
+        proposal.account.openbookTwapFailMarket,
+        proposal.account.baseVault,
+        proposal.account.quoteVault,
+      ]);
+    if (!accountInfos || accountInfos.indexOf(null) >= 0) return;
+
+    const [passOpenBookMarket, failOpenBookMarket] =
+      await Promise.all<MarketAccount>([
+        // pass market is index 0
+        this.openbook.coder.accounts.decode("market", accountInfos[0]!.data),
+        // fail market is index 1
+        this.openbook.coder.accounts.decode("market", accountInfos[1]!.data),
+      ]);
+
+    const bookAccountInfos =
+      await this.rpcProvider.connection.getMultipleAccountsInfo([
+        passOpenBookMarket.asks,
+        passOpenBookMarket.bids,
+        failOpenBookMarket.asks,
+        failOpenBookMarket.bids,
+      ]);
+    const passAsksNodes = getLeafNodes(
```

Comment:
This stuff is deprecated with the latest open book client where they have enriched the capacity for order books and retrieval. So we'll want to update ALL of this section to start to leverage it vs taking the old code.